trustwallet/wallet-core
# fixed duplicate symbol with another lib

## Description

This fix addresses the duplicate symbol issue related to the `blake2b` function, which occurred when linking both `libTrezorCrypto` and `libQt6Core`. The duplication involved symbols such as `'_blake2b'` and `'_blake2b_init_param'`.

`duplicate symbol '_blake2b' in:
    /Users/andrei.haladok/Desktop/work_folder/Libraries/local/Qt-6.2.9/lib/libQt6Core_debug.a[x86_64][123](qcryptographichash.cpp.o)
    /Users/andrei.haladok/Desktop/work_folder/Libraries/local/lib/libTrezorCrypto.a[43](blake2b.c.o)
duplicate symbol '_blake2b_init_param' in:
    /Users/andrei.haladok/Desktop/work_folder/Libraries/local/Qt-6.2.9/lib/libQt6Core_debug.a[x86_64][123](qcryptographichash.cpp.o)
    /Usersandrei.haladokDesktop/work_folder/Libraries/local/lib/libTrezorCrypto.a[43](blake2b.c.o)
ld: 2 duplicate symbols
clang++: error: linker command failed with exit code 1 (use -v to see invocation)`

### What was changed:
In the `blake2b.h` header file of the `libTrezorCrypto` library, a `tc_` prefix was added to `blake2b`-related function names to avoid symbol conflicts:
- `blake2b_Init` -> `tc_blake2b_Init`
- `blake2b_InitKey` -> `tc_blake2b_InitKey`
- `blake2b_InitPersonal` -> `tc_blake2b_InitPersonal`
- `blake2b_Update` -> `tc_blake2b_Update`
- `blake2b_Final` -> `tc_blake2b_Final`
- `blake2b` -> `tc_blake2b`
- `blake2b_Personal` -> `tc_blake2b_Personal`
- `blake2b_Key` -> `tc_blake2b_Key`

These changes were applied throughout the code to ensure the functions align with the updated naming, thus preventing compilation conflicts.

### Why this approach is better:
In my opinion, this change is a cleaner solution compared to using "dirty hacks" that might lead to undefined behavior, security vulnerabilities, or maintenance issues. Alternative workarounds, such as using the following linker flags and hacks, can often obscure the underlying problem and introduce other risks:

- `-unexported_symbols_list`
- `-Xlinker -multiply_defined -Xlinker suppress`
- `-allow_multiply_defined_symbols`
- `-Wl,-dead_strip`
- `-Wl,-force_load`

With this fix, we can avoid manipulating symbols and hiding them at the linking stage, which ensures a more stable and maintainable codebase.

## How to test

1. Navigate to `samples/cpp/CMakeLists.txt`, which is a sample project configuration.
2. Add `libQt6Core_debug` as a dependency in the CMake configuration to replicate the setup where `libQt6Core` and `libTrezorCrypto` are linked together.
3. Build the project and check for any linker errors related to duplicate symbols.
4. Ensure that the project compiles successfully without duplicate symbol errors, confirming that the `tc_` prefixes in `libTrezorCrypto` resolve the conflict with `libQt6Core`.

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

## Checklist


If you're adding a new blockchain
